presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -68,6 +68,9 @@
 
     private final DriverContext driverContext;
     private final List<Operator> operators;
+    // this is present only for debugging
+    @SuppressWarnings("unused")
+    private final List<Operator> allOperators;
```

Comment:
If we want to hold on to the operators for now (for any reason), an index to the operators list would be better. Maybe, `firstUnfinishedOperatorIndex`.

If people see that operator get purged from the `operators` field in driver, people will assume that they become garbage right away (which is something we've been wanting). However, it is not true with your `allOperators` field. As a result, this unused "for-debugging" field can have significant side effect that is non obvious.

Reply:
Let's keep this. But rename `operators` to `activeOperators`